app/uk/gov/hmrc/tai/service/IncomeService.scala

```diff
@@ -89,11 +91,16 @@ class IncomeService @Inject()(
 
   def taxCodeIncomes(nino: Nino, year: TaxYear)(implicit hc: HeaderCarrier): Future[Seq[TaxCodeIncome]] = {
     val eventualIncomes = incomeRepository.taxCodeIncomes(nino, year)
-    val eventualEmployments = employmentService.employments(nino, year).recover { case ex =>
-      Logger.warn(s"EmploymentService.employments - failed to retrieve employments: ${ex.getMessage}")
-      Seq.empty[Employment]
-    }
-
+    val eventualEmployments =
+      if (config.employmentsStatusEnabled) {
+        employmentService.employments(nino, year).recover { case ex =>
+          Logger.warn(s"EmploymentService.employments - failed to retrieve employments: ${ex.getMessage}")
+          Seq.empty[Employment]
```

Comment:
Why do we want to return a Seq.empty[Employment]?

Reply:
Because the method expects: Future[Seq[TaxCodeIncome]] and the type mandates an empty list.
If the list is empty then this:
val map = employments.groupBy(_.sequenceNumber)
Will return an empty map defaulting this: 
employmentStatus.getOrElse(taxCode.status)
to the taxCode.status